lib/hamster/hash.rb

```diff
@@ -590,6 +590,27 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `hil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return value
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      else
+        if value.respond_to?(:dig)
+          value.dig(*rest)
+        end
+      end
+    end
```

Comment:
I would follow the behaviour we currently have in `#update_in` (which shares some ideas with `#dig`):

```
2.2.3 :001 > v = Hamster::Vector[1,2,3]
 => Hamster::Vector[1, 2, 3] 
2.2.3 :002 > v.update_in(1) { |v| 100+v }
 => Hamster::Vector[1, 102, 3] 
2.2.3 :003 > v
 => Hamster::Vector[1, 2, 3] 
2.2.3 :004 > v.update_in(:foo) { |v| 100+v }
ArgumentError: comparison of Symbol with -3 failed
	from /home/dmurik/world/hamster/lib/hamster/vector.rb:261:in `>='
	from /home/dmurik/world/hamster/lib/hamster/vector.rb:261:in `fetch'
	from /home/dmurik/world/hamster/lib/hamster/associable.rb:41:in `update_in'
	from (irb):4
	from /home/dmurik/.rvm/rubies/ruby-2.2.3/bin/irb:11:in `<main>'
```

So, yes, raise an exception when accessing a Vector with a non-integer index.